File: Code_c#_Lab_1/Code_c#_Lab_1/CalculateNum.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Code_c__Lab_1
{
    public class CalculateNum
    {
        private readonly int threadId;
        private readonly BreakThread breakThread;
        private readonly int additionValue;
        private readonly Thread thread;

        public CalculateNum(int threadId, BreakThread breakThread, int[] additionsArray)
        {
            this.threadId = threadId;
            this.breakThread = breakThread;
            additionValue = additionsArray[threadId];
            thread = new Thread(new ThreadStart(Run));
        }

        public void Start()
        {
            thread.Start();
        }
        public void Run()
        {
            long sum = 0;
            long numberOfTerms = 0;
            while (!breakThread.CanBreak)
            {
                sum += additionValue;
                numberOfTerms++;
            }
            Console.WriteLine($"Thread id: {threadId} Sum: {sum} Number of terms: {numberOfTerms} Addition Value: {additionValue}");
        }
    }
}

File: Code_java_Lab_1/untitled/src/CalculateNum.java
public class CalculateNum extends Thread {
    private final int threadId;
    private final BreakThread breakThread;
    private final int additionValue;

    public long getThreadId() {
        return threadId;
    }

    public int getAdditionValue() {
        return additionValue;
    }

    public CalculateNum(int threadId, BreakThread breakThread, int[] additionsArray) {
        this.threadId = threadId;
        this.breakThread = breakThread;
        this.additionValue = additionsArray[threadId];
    }

    @Override
    public void run() {
        long sum = 0;
        long numberOfTerms = 0;
        while (!breakThread.isCanBreak()) {
            sum += additionValue;
            numberOfTerms++;
        }
        System.out.println("Thread id: " + getThreadId() + " Sum: " + sum + " Number of terms: " + numberOfTerms + " Addition Value " + getAdditionValue());
    }
}
